jetty-webapp/src/main/java/org/eclipse/jetty/webapp/Configurations.java

```diff
@@ -409,9 +409,24 @@ private void addConfiguration(Configuration configuration)
             _configurations.add(configuration);
             return;
         }
-
-        if (!_configurations.stream().map(c->c.getClass().getName()).anyMatch(n->{return name.equals(n);}))
+        else
```

Comment:
It is not an else here.  A configuration can both replace another and itself be replaced.

Consider `A` replaces `B` replaces `C`.  `A` and `C` can be added to the configurations list without interaction.  But when `B` is added, we should first detect that it replaces `C` and remove that from the configurations. We should then detect that `A` replaces `B` and not actually add `B`, so we are left with only `A`